_posts/news/2024-03-19-announcing-scalajs-1.16.0.md

```diff
@@ -0,0 +1,98 @@
+
+<!--more-->
+
+## Getting started
+
+If you are new to Scala.js, head over to [the tutorial]({{ BASE_PATH }}/tutorial/).
+
+If you need help with anything related to Scala.js, you may find our community [in `#scala-js` on Discord](https://discord.com/invite/scala) and [on Stack Overflow](https://stackoverflow.com/questions/tagged/scala.js).
+
+Bug reports can be filed [on GitHub](https://github.com/scala-js/scala-js/issues).
+
+## Release notes
+
+If upgrading from Scala.js 0.6.x, make sure to read [the release notes of Scala.js 1.0.0]({{ BASE_PATH }}/news/2020/02/25/announcing-scalajs-1.0.0/) first, as they contain a host of important information, including breaking changes.
+
+This is a **minor** release:
+
+* It is backward binary compatible with all earlier versions in the 1.x series: libraries compiled with 1.0.x through 1.15.x can be used with 1.16.0 without change.
+* It is *not* forward binary compatible with 1.15.x: libraries compiled with 1.16.0 cannot be used with 1.15.x or earlier.
+* It is *not* entirely backward source compatible: it is not guaranteed that a codebase will compile *as is* when upgrading from 1.15.x (in particular in the presence of `-Xfatal-warnings`).
+
+As a reminder, libraries compiled with 0.6.x cannot be used with Scala.js 1.x; they must be republished with 1.x first.
+
+## Enhancements with compatibility concerns
+
+### Changes to the IR and linker APIs
+
+For tooling authors who directly manipulate the IR and linker APIs, there have been some breaking changes in that area.
+This is in line with our version policy for the linker APIs.
+
+The most likely changes you may hit are:
+
+* `FieldName` is now a composite of the defining `ClassName` and a `SimpleFieldName`.
+* `StoreModule` has no parameter anymore; it implicitly applies to storing the `this` value of the enclosing module class.
+
+## Enhancements
+
+### Scala.js-specific minifier
+
+In the `fullLink` mode, the Scala.js linker now includes dedicated code size optimizations.
+They are mostly useful when the Google Closure Compiler is deactivated, which is the case by default when emitting ECMAScript modules.
+
+The most important optimization is that it compresses all *property* names (fields and methods) of Scala classes.
+It computes how frequently each one is used across the codebase, and assigns shorter names to the most frequent ones.
+
+Renaming local and global variables and functions in this way is a standard feature of general-purpose JavaScript minifiers.
+However, in general they cannot rename *properties* without breaking the semantics of JavaScript programs.
+This is why Scala.js has leveraged Closure since the very first relase.
+Closure's so-called "advanced optimizations" assume that the JavaScript program fits in a specific subset of semantics, which allows it to rename properties.
+
+Unfortunately, despite ECMAScript modules being standard for several years now, Closure has never supported them in the way we need it to.
```

Comment:
I think this sentence is a bit misleading. IIUC we could use GCC with CJS modules because it *doesn't* support (i.e. ignores) them.

Reply:
I rephrased. Is it better now?